flopy/modflow/mfflwob.py

```diff
@@ -314,3 +350,230 @@ def write_file(self):
+                    layer.append(k)
+                    row.append(i)
+                    column.append(j)
+                    factor.append(fac)
+                    break
+
+            nobs += 1
+            if nobs == nqfb:
+                break
+
+        irefsp = np.array(irefsp) - 1
+        layer = np.array(layer) - 1
+        row = np.array(row) - 1
+        column = np.array(column) - 1
+        factor = np.array(factor)
+
+        # close the file
+        f.close()
+
+        # get ext_unit_dict if none passed
+        if ext_unit_dict is None:
+            namefile = os.path.join(model.model_ws, model.namefile)
+            ext_unit_dict = parsenamefile(namefile, model.mfnam_packages)
+
+        flowtype, ftype = _get_ftype_from_filename(f.name, ext_unit_dict)
+
+        # set package unit number
+        unitnumber = None
+        filenames = [None, None]
+        if ext_unit_dict is not None:
+            unitnumber, filenames[0] = \
+                model.get_ext_dict_attr(ext_unit_dict,
+                                        filetype=ftype.upper())
+            if iufbobsv > 0:
+                _, filenames[1] = \
+                    model.get_ext_dict_attr(ext_unit_dict, unit=iufbobsv)
+                model.add_pop_key_list(iufbobsv)
+
+        # create ModflowFlwob object instance
+        flwob = ModflowFlwob(model, iufbobsv=iufbobsv, tomultfb=tomultfb,
+                             nqfb=nqfb, nqcfb=nqcfb,
+                             nqtfb=nqtfb, nqobfb=nqobfb, nqclfb=nqclfb,
+                             obsnam=obsnam, irefsp=irefsp, toffset=toffset,
+                             flwobs=flwobs, layer=layer, row=row,
+                             column=column, factor=factor, options=options,
+                             flowtype=flowtype, unitnumber=unitnumber,
+                             filenames=filenames)
+
+        return flwob
+
+
+def _get_ftype_from_filename(fn, ext_unit_dict=None):
+    """
+    Returns the boundary flowtype and filetype for a given ModflowFlwob
+    package filename.
+
+    Parameters
+    ----------
+    fn : str
+        The filename to be parsed.
+    ext_unit_dict : dictionary, optional
+        If the arrays in the file are specified using EXTERNAL,
+        or older style array control records, then `f` should be a file
+        handle.  In this case ext_unit_dict is required, which can be
+        constructed using the function
+        :class:`flopy.utils.mfreadnam.parsenamefile`.
+
+    Returns
+    -------
+    flowtype : str
+        Corresponds to the type of the head-dependent boundary package for
+        which observations are desired (e.g. "CHD", "GHB", "DRN", or "RIV").
+    ftype : str
+        Corresponds to the observation file type (e.g. "CHOB", "GBOB",
+        "DROB", or "RVOB").
+    """
+
+    ftype = None
+
+    # determine filetype from filename using ext_unit_dict
+    if ext_unit_dict is not None:
+        for _, v in ext_unit_dict.items():

```

Comment:
I would use `key` instead of `_` just so it is clear what the first value is.

Reply:
Catch dict items explicitly. New commit pushed to address this one.